fem/eltrans.hpp

```diff
@@ -46,7 +46,15 @@ class ElementTransformation
    const DenseMatrix &EvalInverseJ();
 
 public:
-   int Attribute, ElementNo;
+   enum IndexType
```

Comment:
I chose "index" because the `GridFunction::GetValue` methods used an integer index "i" which was ambiguous because we weren't sure what group of elements was being referenced.  Of course, this PR removes that index so the term "IndexType" is no longer relevant.  This didn't occur to me at the time but you are absolutely correct.

I'm open to suggestions.  `ElementType` would fit but we already use this elsewhere.  On the other hand we don't actually use this as a type so we could make this an anonymous enumeration.  Of course, I should also add comments explaining how it is used.